code/ui/blocks/src/blocks/Source.tsx

```diff
@@ -162,18 +206,22 @@ export const useSourceProps = (
 
         // NOTE: args *does* have to be defined here due to the null check on story above
         const [args] = argsFromStories[index] || [];
+        const storyContext = docsContext.getStoryContext(story);
 
         // eslint-disable-next-line no-underscore-dangle
-        const argsForSource = props.__forceInitialArgs
-          ? docsContext.getStoryContext(story).initialArgs
-          : args;
+        const argsForSource = props.__forceInitialArgs ? storyContext.initialArgs : args;
 
         const source = getStorySource(story.id, argsForSource, sourceContext);
         if (index === 0) {
           // Take the format from the first story
           format = source.format ?? story.parameters.docs?.source?.format ?? false;
         }
-        return getSnippet(source.code, story, props.type);
+        return getSnippet({
+          snippet: source.code,
+          storyContext: { ...storyContext, args: argsForSource },
```

Comment:
@tmeasday I'm a bit unsure about this, whether or not `args` should just always be `args`, or if it should respect `__forceInitialArgs`. I've opted for the latter here, since that makes it easier to do the right thing, but it's technically incorrect.